sdk/nodejs/tests/stackReference.spec.ts

```diff
@@ -70,4 +70,17 @@ describe("StackReference.getOutputDetails", () => {
             secretValue: "supersecretpassword",
         });
     });
+
+    it("types applies correctly", async () => {
+        const passwordValue = "supersecretpassword";
+        pulumi.runtime.setMocks(
+            new TestMocks({
+                password: pulumi.secret(passwordValue),
+            }),
+        );
+        const ref = new pulumi.StackReference("foo");
+        const password: pulumi.Output<number> = ref.outputs["password"].apply(x => x.length);
```

Comment:
Is it worth using `satisfies` to check the type of `ref.outputs["password"]` here as well?

Reply:
Satifies relies on a TypeScript version newer than we use in build, IIRC.